source/ScrollBar.cpp

```diff
@@ -0,0 +1,151 @@
+/* ScrollBar.cpp
+Copyright (c) 2024 by Daniel Yoon
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "ScrollBar.h"
+
+#include "LineShader.h"
+
+#include <algorithm>
+
+namespace {
+	// Additional distance the scrollbar's tab can be selected from.
+	constexpr int SCROLLBAR_MOUSE_ADDITIONAL_RANGE = 5;
+}
+
+ScrollBar::ScrollBar(
+	float fraction,
+	float displaySizeFraction,
+	const Point &from,
+	const Point &to,
+	float tabWidth,
+	float lineWidth,
+	Color color,
+	Color innerColor
+) noexcept :
+	fraction(fraction),
+	displaySizeFraction(displaySizeFraction),
+	from(from),
+	to(to),
+	tabWidth(tabWidth),
+	lineWidth(lineWidth),
+	color(color),
+	innerColor(innerColor)
+{
+}
+
+
+
+ScrollBar::ScrollBar() noexcept : ScrollBar(
+	0.0,
+	0.0,
+	Point(),
+	Point(),
+	3,
+	3,
+	Color(0.6),
+	Color(0.25)
```

Comment:
These should probably be defined in an interface

Reply:
The scrollbars might change colours depending on the type of panel, so it's up to the respective panels to style them.